### ISSUE TITLE
Removing Event Handler doesn't seem to work

### ISSUE THREAD

**Issue by CreepGin** [sebastienros/jint#1139]
`remove_EventName(handler)` doesn't really remove the handler. Take an existing unit test for example:

```C#
public void ShouldExecuteActionCallbackOnEventChanged()
{
    var collection = new System.Collections.ObjectModel.ObservableCollection<string>();
    Assert.True(collection.Count == 0);

    _engine.SetValue("collection", collection);

    RunTest(@"
        var eventAction = 1;
        collection.add_CollectionChanged(handler);
        collection.remove_CollectionChanged(handler);    // <- does not remove the original handler
        collection.Add('test');

        function handler(sender, eventArgs) { eventAction = 0 }
    ");

    var eventAction = _engine.GetValue("eventAction").AsNumber();
    Assert.Equal(1, eventAction);    // Throws. Handler is still called
}
```

**Comment by CreepGin** [comment]
After thinking it over again, I think there's not much Jint can do in this case, right? The interop makes it hard to support. Users may need to rely on other workarounds.

**Comment by lahma** [comment]
I've created a PR that should address this, #1146 . Waiting from feedback whether it's safe to cache the delegate to keep same identity and thus fixing this use case.

**Comment by CreepGin** [comment]
@lahma That is awesome! Ty for taking the time to address this. I spent 1.5 days tackling this from outside of Jint. Ended up with a `CreateDelegate` approach that works but depends on a dedicated Event Manager for wrapping delegates and adding/removing events. Needless to say, a delegate cache from within Jint is much better. And, I just realized that I need to get comfortable with Linq.Expressions.
